core/models.py

```diff
@@ -58,26 +57,26 @@ class Question(models.Model):
     question_text = RichTextUploadingField()
     choice1 = RichTextUploadingField()
     choice2 = RichTextUploadingField()
-    choice3= RichTextUploadingField()
+    choice3 = RichTextUploadingField()
     choice4 = RichTextUploadingField()
     correct_choice = models.PositiveIntegerField(blank=False)
     negative = models.BooleanField(default=False)
     negative_marks = models.IntegerField(null=True, blank=True)
     marks = models.IntegerField(null=True, default=1)
 
-
     def __str__(self):
         return self.question_text
 
 
 class Candidate(models.Model):
     name = models.CharField(max_length=100, blank=False)
-    # std_no_regex = RegexValidator(regex=r"^\d{7}$")
-    std_no = models.CharField(blank=True, max_length=7, null=True)
-    university_roll_no = models.CharField(max_length=10, unique=True)
+    std_no_regex = RegexValidator(regex=r"^\d{7}$", message="Invalid Student Number", code="400")
+    std_no = models.CharField(validators=[std_no_regex], blank=True, max_length=7, null=True)
+    university_roll__no_regex = RegexValidator(regex=r'^[1]\d{9}$',message="invalid university roll no.")
+    university_roll_no = models.CharField(max_length=10,blank=True, unique=True,validators=[university_roll__no_regex],null=True)
     email = models.EmailField(unique=True, blank=False)
     father = models.CharField(max_length=255, blank=True, null=True)
-    phone_regex = RegexValidator(regex=r"^[789]\d{9}$")
+    phone_regex = RegexValidator(regex=r"^[6789]\d{9}$")
```

Comment:
This regex won't accept all kinds of valid phone numbers. Now numbers start from 5 also. Use a valid regex.

Reply:
I improved the previous ones and added 6. Ok I will add 5 also

Reply:
search for a better regex instead of just adding 5.

Reply:
> This regex won't accept all kinds of valid phone numbers. Now numbers start from 5 also. Use a valid regex.

Sir no number starts from 5 at least not this year. So I think for security purpose this is ok.